[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs

sanrise opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139837

Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc @robieta @chaekit @guotuofeng @guyang3532 @dzhulgakov @davidberard98 @briancoutinho @sraikund16

sanrise committed:
[pytorch/profielr] Profiler NCCL metadata can now contain start and end addr hints for Tensor values.

Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead on custom silicon.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

[ghstack-poisoned]

sanrise commented:
@pytorchbot label "topic: not user facing"

sanrise committed:
Update on "[pytorch/profielr] Profiler NCCL metadata can now contain start and end addr hints for Tensor values."


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead on custom silicon.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain start and end addr hints for Tensor values."


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain start and end addr hints for Tensor values."


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sraikund16 reviewed (approved):
discussed concerns offline. Looks like it is very light-weight especially since we only annotate the CPU event

future steps may include:

1.frontend python interface to turn off this feature (and maybe for on-demand as well?)
2.Propagate args to kernel events in kineto if necessary

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise committed:
Update on "[pytorch/profiler] Profiler NCCL metadata can now contain collective Input and Ouput Tensor addrs"


Studying memory access patterns is the primary use cases.

Internal: The data may be used to find the % of operators that may cause alignment related overhead.

Differential Revision: [D64413699](https://our.internmc.facebook.com/intern/diff/D64413699/)

cc robieta chaekit guotuofeng guyang3532 dzhulgakov davidberard98 briancoutinho sraikund16

[ghstack-poisoned]

sanrise commented:
This diff was reverted due to a LOG statement, I am grafting a fix and will submit a new PR.
